### PR TITLE
feat: add gosec linter and fix warnings

### LINKED ISSUE #350: Add security code checking using gosec linter
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
Scan source code for security vulnerabilities using gosec.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):
- Kubernetes version: (use `kubectl version`):

### PR DESCRIPTION
**Pull request by aramase**
**What this PR does / why we need it**:
- Adds `gosec`, `deadcode`, `structcheck` and `varcheck` linters to golangci config.
- Switch to using sha256 as sha1 is weak

```bash
pkg/util/secretutil/secret.go:22:2: G505: Blocklisted import crypto/sha1: weak cryptographic primitive (gosec)
        "crypto/sha1"
        ^
pkg/util/secretutil/secret.go:201:12: G401: Use of weak cryptographic primitive (gosec)
        hasher := sha1.New()
```

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #350 

**Special notes for your reviewer**:

### CODE REVIEW

File: test/bats/vault.bats [hunk 1]
```diff
@@ -79,13 +79,13 @@ EOF
...
 
-  run kubectl exec -it $VAULT_POD -- vault write auth/kubernetes/config \
+  run kubectl exec $VAULT_POD -- vault write auth/kubernetes/config \
     kubernetes_host="${K8S_HOST}" \
     kubernetes_ca_cert="${K8S_CACERT}" \
     token_reviewer_jwt="${TR_ACCOUNT_TOKEN}"
   assert_success
 
-  run kubectl exec -it $VAULT_POD -- vault policy write example-readonly -<<EOF
+  run kubectl exec -ti $VAULT_POD -- vault policy write example-readonly -<<EOF
```
Comment (ritazh) [inline test/bats/vault.bats:88]: do we still need `-it` here?

Comment (aramase) [reply test/bats/vault.bats:88]: yeah, this is still required. Without `-it` the multiline command fails based on testing locally.

**Pull request merged**